From cfee89519014f5b57d7eb345e592bba4a6dd8e3a Mon Sep 17 00:00:00 2001
From: Khalid Faisal Ansari <khalid.ansari@oss.qualcomm.com>
Date: Tue, 2 Dec 2025 10:51:13 +0530
Subject: [PATCH 2/2] FROMLIST: arm64: dts: qcom: qcs8275-iq-8275-evk: Enable
 TPM

Enable ST33HTPM TPM over SPI10 on the Monaco IoT EVK by adding the
required SPI and TPM nodes.

Signed-off-by: Khalid Faisal Ansari <khalid.ansari@oss.qualcomm.com>
Upstream-Status: Submitted [https://lore.kernel.org/all/20251114-enable-tpm-monaco-v1-1-4ed566838698@oss.qualcomm.com/]
---
 arch/arm64/boot/dts/qcom/qcs8275-iq-8275-evk.dts | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/qcs8275-iq-8275-evk.dts b/arch/arm64/boot/dts/qcom/qcs8275-iq-8275-evk.dts
index 4660178b42c8..f459de1484de 100644
--- a/arch/arm64/boot/dts/qcom/qcs8275-iq-8275-evk.dts
+++ b/arch/arm64/boot/dts/qcom/qcs8275-iq-8275-evk.dts
@@ -418,6 +418,16 @@ vbus_supply_regulator2: vbus-supply-regulator2 {
 	};
 };
 
+&spi10 {
+	status = "okay";
+
+	tpm@0 {
+		compatible = "st,st33htpm-spi", "tcg,tpm_tis-spi";
+		reg = <0>;
+		spi-max-frequency = <20000000>;
+	};
+};
+
 &tlmm {
 
 	dp_hot_plug_det: dp-hot-plug-det-state {
-- 
2.34.1

